src/ucm/mmap/install.c

```diff
@@ -69,66 +84,160 @@ static void ucm_mmap_event_test_callback(ucm_event_type_t event_type,
     *out_events |= event_type;
 }
 
-void ucm_fire_mmap_events(int events)
+/* Fire events with pre/post callback. The problem is in call sequence: we
+ * can't just fire single event - most of the system calls require set of
+ * calls to eliminate resource leaks or data corruption, such sequence
+ * produces additional events which may affect to event. To exclude
+ * additional events from processing used pre/post callbacks where
+ * handler may be [un]registered only for specific system call */
+static void
+ucm_fire_mmap_events_with_cb(int events, ucm_fire_mmap_cb_t cb, void *arg)
 {
     int shmid;
     void *p;
 
     if (events & (UCM_EVENT_MMAP|UCM_EVENT_MUNMAP|UCM_EVENT_MREMAP|
                   UCM_EVENT_VM_MAPPED|UCM_EVENT_VM_UNMAPPED)) {
-        p = mmap(NULL, 0, 0, 0, -1 ,0);
-        p = mremap(p, 0, 0, 0);
-        munmap(p, 0);
+        UCM_FIRE_EVENT(events, UCM_EVENT_MMAP|UCM_EVENT_VM_MAPPED,
```

Comment:
why need to set/unset callback for each test?
can't we just clear the state of "received events" before every test?

Reply:
the problem is in external event handlers: OMPI produces only VM_UNMAP events and we can't know which call produced event without setting personal handler per call